bitnami/keycloak/templates/tls-secret.yaml

```diff
@@ -9,16 +9,16 @@ SPDX-License-Identifier: APACHE-2.0
 apiVersion: v1
 kind: Secret
 metadata:
-  name: {{ .name }}
+  name: {{ tpl .name $ }}
   namespace: {{ include "common.names.namespace" $ | quote }}
   labels: {{- include "common.labels.standard" ( dict "customLabels" $.Values.commonLabels "context" $ ) | nindent 4 }}
   {{- if $.Values.commonAnnotations }}
   annotations: {{- include "common.tplvalues.render" ( dict "value" $.Values.commonAnnotations "context" $ ) | nindent 4 }}
   {{- end }}
 type: kubernetes.io/tls
 data:
-  tls.crt: {{ .certificate | b64enc }}
-  tls.key: {{ .key | b64enc }}
+  tls.crt: {{ tpl .certificate $ | b64enc }}
+  tls.key: {{ tpl .key $ | b64enc }}
```

Comment:
I see the value of using `tpl` in `name`, but not really that much here. At the end of the day, it would be strange some kind of templating applied to a certificate or key content, isn't it?

Reply:
Imagine including the chart as a subchart into an umbrella chart. The umbrella chart has several ingresses, all using a wildcard certificate.

In this following example the certificate is for "*.something.com". 

```
global:
  keycloak:
    hostname: keycloak.something.com
  ingress:
    tlsCert: somecert
    tlsCert: somekey


keycloak:
  (...)
  ingress:
    (...)
    hostname: "{{ .Values.global.keycloak.hostname }}"
    secrets:
      - name: "{{ printf \"%s-tls\" .Values.global.keycloak.hostname }}"
        certificate: "{{ .Values.global.ingress.tlsCert }}"
        key: "{{ .Values.global.ingress.tlsKey }}"
```

My goal is that the umbrella chart can be configured with minimal values from the outside. I would like to hide the complex configuration of the individual charts as much as possible.

Reply:
Gotcha! I see your point now, thanks 😄